### DISCUSSION TITLE
Introducing zigbee2mqtt-automations: enhance your zigbee2mqtt experience

### DISCUSSION THREAD

**Discussion by Luligu** [Koenkk/zigbee2mqtt#20078] [Show and tell]
Hello Zigbee2MQTT Community,

I'm excited to share a project I've been working on, https://github.com/Luligu/zigbee2mqtt-automations, which I believe many of you will find useful in your smart home setups.

About the Project:
Zigbee2mqtt-automations is designed to complement your Zigbee2MQTT setup by adding automations directly inside zigbee2mqtt. Whether you're looking for simple or advanced automations, this tool aims to make it easier and more intuitive.

Key Features:
- Support multiple event based triggers;
- Support time automations (execution at specified time);
- Support suncalc automations like sunset, sunrise, dawn, dusk and others at a specified location and altitude;
- Support multiple conditions based on an entity state or attribute or based time;
- Provides comprehensive logging within the zigbee2mqtt logging system for triggers, conditions and actions;
- Performs thorough validation of the automation configuration file for errors (errors are logged at loading time and the erroneous automation is discarded);
- Error messages and execution notifications can be displayed as pop-up messages in frontend.
- User can filter automation events in the frontend by entering [Automations] in the 'Filter by text' field. 

Why You Might Want to Try It:
Automations at zigbee2mqtt level execute at light speed. well before top platforms receive a mqtt message.
All automations are in a simple text file: easy to change and save.

Getting Started:
The project is open-source (MIT license) and available on GitHub. You can find detailed instructions on how to set it up and use it in your own environment. I've put effort into making the documentation clear and helpful, but if you have questions or run into issues, feel free to raise them in the repository's Issues section.

Feedback and Contributions:
I'm looking for feedback to improve and expand the project. Whether it's a feature request, a bug report, or a suggestion for improvement, your input is valuable. Contributions to the code are also welcome if you're inclined to do so!

Join the Journey:
I believe this project has the potential to grow and become an integral part of our Zigbee2MQTT experiences. Your support, feedback, and contributions can shape its future.

Thank you for checking out zigbee2mqtt-automations, and I can't wait to see how it helps in your smart home adventures!

Best regards,
Luligu

**Comment by mundschenk-at** [comment]
I haven't yet been able to experiment with your extension, but it sounds like a fantastic idea and looks to be quite full-featured for a first release!

**Comment by chris-1243** [comment]
I just had a try and it is a fantastic job....

**Reply by Luligu** [reply]
Thanks

**Comment by haeewee** [comment]
Thanks!  looks  very  promising.  I like it a lot  the  automation  is  very  near to the devices !

**Reply by Luligu** [reply]
Thanks! That is the reason I wrote it and use it.

**Comment by Luligu** [comment]
Happy new year to everyone.

**Comment by domcars0** [comment]
Hi, happy new year and thank you for this work. 
I'm trying to transform a 2 gang switch to a curtain switch. So all I want is when a gang is set to ON, the other must be set to OFF. I write this automations.yaml but it does not works. Could you help me please? Thanks
The  'Store Switch' (name) attributs are: 
{
    "last_seen": "2025-01-10T12:30:27.537Z",
    "linkquality": 138,
    "power_on_behavior": "previous",
    "power_on_behavior_l1": "previous",
    "state_l1": "ON",
    "state_l2": "OFF",
    "switch_type": "state"
}


pi@raspbuzi:~ $ cat /opt/zigbee2mqtt/data/automations.yaml                   store_jardin_up:
  active: true
  execute_once: true
  trigger:
    entity: "Store Switch"
    attribut: "state_l2"
    equal: "ON"
  action:
    entity: "Store Switch"
    payload: { "state_l1" : "OFF"}
    logger: debug

store_jardin_down:
  active: true
  execute_once: true
  trigger:
    entity: "Store Switch"
    attribut: "state_l1"
    equal: "ON"
  action:
    entity: "Store Switch"
    payload: { "state_l2" : "OFF"}
    logger: debug

**Comment by gohlas** [comment]
Just found this!
Great idea. Will give a try here!
Thanks for the work

**Comment by OACDesigns** [comment]
this sounds great! 
I would love to have these automations present as a fallback for when/if my HomeAssistant goes down.
Is there a way to externally enable/disable the automations? Perhaps with an mqtt message?
That way I could have a small service checking the health of my HA system and sending an `enable` message to turn on these automations if it's down.

**Reply by gohlas** [reply]
+1 on on this.
It would be very helpfull with a fallback system.

**Comment by Luligu** [comment]
Right now the only external command available is to publish:
zigbee2mqtt-automations/<AUTOMATION_NAME> with raw payload "execute" to manually execute.

We could think to add more external command if they are useful. Like "enable" and "disable".

Btw I'm adding scenes management too. 
They execute at z2m level super fast and can be configured with payload for each single device so much more powerful then anything else.
Once fully tested for a while I will publish it.

Next steps is to add webhooks so we can control almost everything at z2m level (if you have a scene button in z2m from it you will be able for instance to turn on or off a Shelly device!).

**Comment by Luligu** [comment]
Hi,
released version 3 with scenes.

## [3.0.0] - 2025-09-11

### Added

- [scenes]: Added support for scenes.
- [readme]: Added docs for triggering an automation and executing a scene with mqtt.

### Changed

- [package]: Update dependencies.
- [package]: Update package.json.
- [package]: Update tsconfig.json.

<a href="https://www.buymeacoffee.com/luligugithub">
  <img src="bmc-button.svg" alt="Buy me a coffee" width="80">
</a>

**Comment by svrooij** [comment]
@Luligu what a great idea!!! I had a similar idea a few years ago but never got to it!

Quick question does this also mean zigbee2mqtt will also start working when there is no mqtt sever up because of some disturbance/update going on?

Would love to see where this goes!

**Reply by Luligu** [reply]
Thanks, I appreciate it.

The next idea is to add webhook so really all automations scenes can be there.

I didn't try but 3.0.0 injects directly to z2m so I guess should work also with broker down.
